src_c/base.c

```diff
@@ -188,6 +188,44 @@ pg_SetDefaultWindowSurface(PyObject *);
 #endif /* IS_SDLv2 */
 
 
+#ifdef WIN32
+static FILE*
+pg_Fopen(const char *filename, const char *mode) {
+    FILE *fp = NULL;
+    static wchar_t modebuf[24];
+    size_t nameInputSize = strlen(filename) + 1;
+    size_t namebuf_chars = MultiByteToWideChar(CP_UTF8, 0, filename, nameInputSize, 0, 0);
+    wchar_t* namebuf;
+    if (namebuf_chars == 0)
+        return NULL;
+    namebuf = malloc(namebuf_chars * sizeof(wchar_t));
+
+    if (!namebuf)
+        goto leave;
+    if (MultiByteToWideChar(CP_UTF8, 0, filename, nameInputSize,
+                            namebuf, namebuf_chars) == 0) {
+        PyErr_Format(PyExc_OSError, "error code %u", GetLastError());
+        goto leave;
+    }
+    if (MultiByteToWideChar(CP_UTF8, 0, mode, -1,
+                            modebuf, sizeof(modebuf) / sizeof(wchar_t)) == 0) {
+        PyErr_Format(PyExc_OSError, "error code %u", GetLastError());
+        goto leave;
+    }
+
+    fp = _wfopen(namebuf, modebuf);
+
+leave:
+    free(namebuf);
+    return fp;
+}
+#else /* !WIN32 */
+static FILE*
+pg_Fopen(const char *filename, const char *mode) {
+    return fopen(filename, mode);
```

Comment:
I wonder what we have to do about the GIL here? Does it need to be wrapped in Py_BEGIN_ALLOW_THREADS; ?

Reply:
If SDL is used for error handling, then the caller can safely use Py_BEGIN_ALLOW_THREADS. It shouldn't matter if the GIL is held by the thread, but it doesn't have to be held?
Maybe imageext should be rewritten to use `pgRWopsFromObjectThreaded()` or `SDL_RWFromFile()` at some point. Then `pg_Fopen()` won't be needed anymore.